### ISSUE TITLE
Crashes KeePassDX right after start (without interaction)

### ISSUE THREAD

**Issue by Avispa** [zhanghai/MaterialFiles#716]
Not 100% sure where to post this but pretty confident it's the fault of _MaterialFiles_.

After removing "Access" of _MaterialFiles_ from the permissions menu of _KeePassDX_/Android, I could choose which file manager to use in _KeyPassDX_ to select the correct database and from there all went fine.

Since nowerdays every stupid application blocks me from doing screenshots I cannot tell the exact error message anymore but wasn't much informative either. Just a two-liner including the terms "MaterialFiles", "access" and "app crash".

### System
OS: Android 9 (2019-06-01)
Phone: LG v40

**Comment by cordelia04m** [comment]
I have the same issue on KeepassDX 3.3.1. Downgrading to 3.3 resolves this so I think the cash is due to KeepassDX's change.

**Comment by cordelia04m** [comment]
Now that's weird. While I was trying to find a reproducible steps of the bug, it somehow worked as intended. I'm not sure which app is the cause of this but your should report it to keepassDX as well.

**Comment by cordelia04m** [comment]
Update: it crashed again. This bug is hard to reprouce.

**Comment by J-Jamet** [comment]
I just noticed it too, here is a way to reproduce it: 
1. Kill KeePassDX and Material Files applications
2. Putting the device to sleep
3. Open KeePassDX which has a link to an open database with Material Files
4. The problem occurs

Solve the problem temporarily:
1. Start by opening the Material Files application so that an instance is opened in the background
2. Open KeePassDX with a link to a database provided by Material Files
3. Everything is OK, the file can be opened normally

Configuration:
KeePassDX 3.3.2
Material Files 1.4.0 from F-Droid
One Plus One with Lineage OS (no Play Services) Android 10

Logs : 
```
2022-03-22 16:22:28.346 27953-27970/me.zhanghai.android.files E/JavaBinder: *** Uncaught remote exception!  (Exceptions are not yet supported across processes.)
    java.lang.ExceptionInInitializerError
        at me.zhanghai.android.files.provider.linux.LinuxPath.j(Unknown Source:4)
        at fb.t.f(:5)
        at me.zhanghai.android.files.provider.root.RootablePosixFileAttributeView.l(Unknown Source:5)
        at me.zhanghai.android.files.provider.root.RootablePosixFileAttributeView.a(Unknown Source:4)
        at ua.g.v(Unknown Source:45)
        at me.zhanghai.android.files.provider.root.m.v(Unknown Source:26)
        at ii.t0.E(:3)
        at me.zhanghai.android.files.file.FileProvider.query(:2)
        at android.content.ContentProvider.query(ContentProvider.java:1214)
        at android.content.ContentProvider.query(ContentProvider.java:1307)
        at android.content.ContentProvider$Transport.query(ContentProvider.java:267)
        at android.content.ContentProviderNative.onTransact(ContentProviderNative.java:104)
        at android.os.Binder.execTransactInternal(Binder.java:1021)
        at android.os.Binder.execTransact(Binder.java:994)
     Caused by: java.lang.IllegalStateException: Cannot invoke setValue on a background thread
        at androidx.lifecycle.LiveData.a(Unknown Source:21)
        at androidx.lifecycle.LiveData.v(Unknown Source:2)
        at cb.c0.B(:11)
        at cb.c0.<clinit>(Unknown Source:7)
        at me.zhanghai.android.files.provider.linux.LinuxPath.j(Unknown Source:4) 
        at fb.t.f(:5) 
        at me.zhanghai.android.files.provider.root.RootablePosixFileAttributeView.l(Unknown Source:5) 
        at me.zhanghai.android.files.provider.root.RootablePosixFileAttributeView.a(Unknown Source:4) 
        at ua.g.v(Unknown Source:45) 
        at me.zhanghai.android.files.provider.root.m.v(Unknown Source:26) 
        at ii.t0.E(:3) 
        at me.zhanghai.android.files.file.FileProvider.query(:2) 
        at android.content.ContentProvider.query(ContentProvider.java:1214) 
        at android.content.ContentProvider.query(ContentProvider.java:1307) 
        at android.content.ContentProvider$Transport.query(ContentProvider.java:267) 
        at android.content.ContentProviderNative.onTransact(ContentProviderNative.java:104) 
        at android.os.Binder.execTransactInternal(Binder.java:1021) 
        at android.os.Binder.execTransact(Binder.java:994) 
2022-03-22 16:22:28.347 27953-27970/me.zhanghai.android.files E/AndroidRuntime: FATAL EXCEPTION: Binder:27953_1
    Process: me.zhanghai.android.files, PID: 27953
    java.lang.ExceptionInInitializerError
        at me.zhanghai.android.files.provider.linux.LinuxPath.j(Unknown Source:4)
        at fb.t.f(:5)
        at me.zhanghai.android.files.provider.root.RootablePosixFileAttributeView.l(Unknown Source:5)
        at me.zhanghai.android.files.provider.root.RootablePosixFileAttributeView.a(Unknown Source:4)
        at ua.g.v(Unknown Source:45)
        at me.zhanghai.android.files.provider.root.m.v(Unknown Source:26)
        at ii.t0.E(:3)
        at me.zhanghai.android.files.file.FileProvider.query(:2)
        at android.content.ContentProvider.query(ContentProvider.java:1214)
        at android.content.ContentProvider.query(ContentProvider.java:1307)
        at android.content.ContentProvider$Transport.query(ContentProvider.java:267)
        at android.content.ContentProviderNative.onTransact(ContentProviderNative.java:104)
        at android.os.Binder.execTransactInternal(Binder.java:1021)
        at android.os.Binder.execTransact(Binder.java:994)
     Caused by: java.lang.IllegalStateException: Cannot invoke setValue on a background thread
        at androidx.lifecycle.LiveData.a(Unknown Source:21)
        at androidx.lifecycle.LiveData.v(Unknown Source:2)
        at cb.c0.B(:11)
        at cb.c0.<clinit>(Unknown Source:7)
        at me.zhanghai.android.files.provider.linux.LinuxPath.j(Unknown Source:4) 
        at fb.t.f(:5) 
        at me.zhanghai.android.files.provider.root.RootablePosixFileAttributeView.l(Unknown Source:5) 
        at me.zhanghai.android.files.provider.root.RootablePosixFileAttributeView.a(Unknown Source:4) 
        at ua.g.v(Unknown Source:45) 
        at me.zhanghai.android.files.provider.root.m.v(Unknown Source:26) 
        at ii.t0.E(:3) 
        at me.zhanghai.android.files.file.FileProvider.query(:2) 
        at android.content.ContentProvider.query(ContentProvider.java:1214) 
        at android.content.ContentProvider.query(ContentProvider.java:1307) 
        at android.content.ContentProvider$Transport.query(ContentProvider.java:267) 
        at android.content.ContentProviderNative.onTransact(ContentProviderNative.java:104) 
        at android.os.Binder.execTransactInternal(Binder.java:1021) 
        at android.os.Binder.execTransact(Binder.java:994)
```

**Comment by zhanghai** [comment]
I see, this is the top issue I observed on Crashlytics as well and it has been fixed in 9a4d287. I can do a patch release this week to fix this issue.
